src/dynamic_fmt.c

```diff
@@ -380,12 +383,13 @@ char *RemoveHEX(char *output, char *input)
 	*cpo++ = *cpi;
 	cpi += 5;
 	while (*cpi) {
-		if (*cpi == '0' && cpi[1] == '0') {
-			strcpy(output, input);
-			return output;
-		}
 		if (atoi16[ARCH_INDEX(*cpi)] != 0x7f && atoi16[ARCH_INDEX(cpi[1])] != 0x7f) {
-			*cpo++ = atoi16[ARCH_INDEX(*cpi)]*16 + atoi16[ARCH_INDEX(cpi[1])];
+			char c = atoi16[ARCH_INDEX(*cpi)]*16 + atoi16[ARCH_INDEX(cpi[1])];
+			if ((only_null && c == 0) || (!only_null && (c < ' ' || c > 0x7e))) {
+				strcpy(output, input);
```

Comment:
How do we know(?) we won't overflow `output` here?

Reply:
The caller is expected (from original code) to supply an output buffer at least as big as the input buffer and the result from "maybe removing hex" can never only ever be shorter. As far as I understand the callees do the right thing although I'm slightly wary about one that use a buffer of 512 instead of the ususal 1024. I just assumed Jim happened to know what he was smoking at that particular moment 😆